src/parser/SpatialQuery.cpp

```diff
@@ -207,10 +207,11 @@ SpatialJoinConfiguration SpatialQuery::toSpatialJoinConfiguration() const {
 
 // ____________________________________________________________________________
 SpatialQuery::SpatialQuery(const SparqlTriple& triple) {
-  AD_CONTRACT_CHECK(triple.p_.isIri(),
+  AD_CONTRACT_CHECK(std::holds_alternative<PropertyPath>(triple.p_) &&
+                        std::get<PropertyPath>(triple.p_).isIri(),
```

Comment:
with the function that returns `optional<string>` for the iri, these three lines (including below) also become simpler.